xarray/backends/pynio_.py

```diff
@@ -28,6 +29,17 @@ def get_array(self):
         return self.datastore.ds.variables[self.variable_name]
 
     def __getitem__(self, key):
+        if isinstance(key, indexing.VectorizedIndexer):
+            raise NotImplementedError(
+                'Nio backend does not support vectorized indexing. Load your '
+                'data first with .load() or .compute(). Given {}'.format(key))
+        # Nio only supports slice indexing, not an array
+        if (isinstance(key, indexing.OuterIndexer) and
+                not all(isinstance(k, (integer_types, slice)) for k in key)):
```

Comment:
We should probably clarify on the indexer classes, but my understanding was:

- `BasicIndexer`: every element is an integer or slice
- `OuterIndexer`: every element is an integer, slice or 1d integer array
- `VectorizedIndexer`: every element is an integer, slice or N-dimensional integer array

Both outer and vectorized indexing generalize basic indexing, but in different ways.

So I think for this case we should already only be using a basic indexer.

Reply:
You are right.
I don't remember why I did that. Updated.